iot-hub/Samples/device/DeviceReconnectionSample/readme.md

```diff
@@ -160,7 +160,7 @@ The device client exhibits the following connection status changes with reason:
   <tr>
     <td> Retry_Expired </td>
     <td> This signifies that the client was disconnected due to a transient exception, but the retry policy expired before a connection could be re-established. </td>
-    <td> If you want to perform more operations on the device client, you should dispose and then re-initialize the client. </td>
+    <td> If you want to perform more operations on the device client, you should dispose and then re-initialize the client. </br> Note that the SDK's default retry policy is set to never expire. </td>
```

Comment:
Does the 4 minute default operation timeout not apply here?

Reply:
The retry policy won't expire, your operation will time out -> the client will still be in connected state. With the retry policy expiring, the client would go into disconnected state.

Reply:
Actually let me test it out, it'll be really bad if I put it into the doc and it turns out to be incorrect.

Reply:
Yeah, we get a task cancelled exception - the retry policy is still active:
```
2020-11-11T15:07:50.8091249>> Debug - Connection status changed: status=Disconnected_Retrying, reason=Communication_Error
2020-11-11T15:07:50.8170451>> Debug - ### The DeviceClient is retrying based on the retry policy. Do NOT close or open the DeviceClient instance
2020-11-11T15:07:51.2935904>> Error - A network related exception was caught, but will try to recover and retry explicitly: System.TimeoutException: Timed out to acquire handler lock.
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.EnsureOpenedAsync(TimeoutHelper timeoutHelper)
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.<>c__DisplayClass19_0.<<ReceiveAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.ReceiveAsync(TimeoutHelper timeoutHelper)
   at Microsoft.Azure.Devices.Client.InternalClient.ReceiveAsync(TimeSpan timeout)
   at Microsoft.Azure.Devices.Client.Samples.DeviceReconnectionSample.ReceiveMessageAndCompleteAsync() in C:\MyData\SourceCode\csharp\temp\azure-iot-samples-csharp\iot-hub\Samples\device\DeviceReconnectionSample\DeviceReconnectionSample.cs:line 293
   at Microsoft.Azure.Devices.Client.Samples.DeviceReconnectionSample.ReceiveMessagesAsync(CancellationToken cancellationToken) in C:\MyData\SourceCode\csharp\temp\azure-iot-samples-csharp\iot-hub\Samples\device\DeviceReconnectionSample\DeviceReconnectionSample.cs:line 256
2020-11-11T15:08:01.7451503>> Error - A network related exception was caught, but will try to recover and retry: System.TimeoutException: The operation timed out.
 ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.SendEventAsync(Message message, CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.InternalClient.SendEventAsync(Message message)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Devices.Client.InternalClient.SendEventAsync(Message message)
   at Microsoft.Azure.Devices.Client.Samples.DeviceReconnectionSample.SendMessagesAsync(CancellationToken cancellationToken) in C:\MyData\SourceCode\csharp\temp\azure-iot-samples-csharp\iot-hub\Samples\device\DeviceReconnectionSample\DeviceReconnectionSample.cs:line 213
2020-11-11T15:08:36.8624005>> Error - A network related exception was caught, but will try to recover and retry: System.TimeoutException: The operation timed out.
 ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Microsoft.Azure.Devices.Client.Transport.RetryDelegatingHandler.SendEventAsync(Message message, CancellationToken cancellationToken)
   at Microsoft.Azure.Devices.Client.InternalClient.SendEventAsync(Message message)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Devices.Client.InternalClient.SendEventAsync(Message message)
   at Microsoft.Azure.Devices.Client.Samples.DeviceReconnectionSample.SendMessagesAsync(CancellationToken cancellationToken) in C:\MyData\SourceCode\csharp\temp\azure-iot-samples-csharp\iot-hub\Samples\device\DeviceReconnectionSample\DeviceReconnectionSample.cs:line 213
2020-11-11T15:08:47.8606058>> Debug - Connection status changed: status=Connected, reason=Connection_Ok
2020-11-11T15:08:47.8612025>> Debug - ### The DeviceClient is CONNECTED; all operations will be carried out as normal.
2020-11-11T15:08:48.3162341>> Information - Sent message 3 of {"temperature":29,"humidity":64}
```

This was with an operation timeout of 30secs, disconnection of about 1 min.